plugin/restore/mysql/mysql.go

```diff
@@ -123,6 +132,165 @@ func getPITRDatabaseName(database string, timestamp int64) string {
 	return getSafeName(database, suffix)
 }
 
+// SyncArchivedBinlogFiles syncs the binlogs between the instance and `saveDir`, but exclude latest binlog
+func (r *Restore) SyncArchivedBinlogFiles(ctx context.Context, instance *api.Instance, saveDir string) error {
+	binlogFilesLocal, err := ioutil.ReadDir(saveDir)
+	if err != nil {
+		return err
+	}
+
+	binlogFilesOnServer, err := r.getBinlogFilesMetaOnServer(ctx)
+	if err != nil {
+		return err
+	}
+
+	latestBinlogFileOnServer, err := r.showLatestBinlogFile(ctx)
+	if err != nil {
+		return err
+	}
+
+	// compare file sizes and names to decide which files to download
+	downloadedIndex := make(map[int]struct{})
```

Comment:
nit: Please give an example in the comment showing the relationship between `downloadedIndex` and `binlogFilesOnServer`.

For example, if `binlogFilesOnServer` is `["binlog.000001", "binlog.000002", "binlog.000003"]`, and `localFile` is `["binlog.0000001"]`, then `downloadedIndex` will be `{0:struct{}}`.